src/deadline/job_attachments/bucket_sweeper/sweeper_processor.py

```diff
@@ -0,0 +1,200 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+import os
+import csv
+
+from typing import List, Dict, Any
+from botocore.exceptions import BotoCoreError
+from botocore.client import BaseClient
+
+from ..exceptions import SweeperProcessorError, JobAttachmentS3BotoCoreError
+
+
+class SweeperProcessor:
+    """Processes cleanup operations for job attachments."""
+
+    def __init__(
+        self,
+        s3_client: BaseClient,
+        s3_control_client: BaseClient,
+        deadline_client: BaseClient,
+        farm_id: str,
+        account_id: str,
+        role_arn: str,
+        bucket_name: str,
+    ):
+        """
+        Initializes the SweeperProcessor.
+
+        Args:
+            s3_client: AWS S3 client for basic S3 operations
+            s3_control_client: AWS S3 Control client for batch operations
+            deadline_client: Client for interacting with Deadline
+            farm_id (str): The target farm_id to cleanup
+            account_id (str): AWS account ID for the batch operation
+            role_arn (str): The ARN of the IAM role for executing batch jobs.
+                Required permissions:
+                    - s3:GetObject
+                    - s3:PutObjectTagging
+                    - s3:CreateJob
+            bucket_name (str): target S3 bucket to cleanup
+        """
+        self.s3 = s3_client
+        self.s3_control = s3_control_client
+        self.deadline = deadline_client
+        self.farm_id = farm_id
+        self.account_id = account_id
+        self.role_arn = role_arn
+        self.bucket_name = bucket_name
+
+    def _create_tag_manifest(self, write_directory: str, delete_list: List[str]) -> str:
+        """
+        Creates a CSV manifest file containing object keys to be deleted.
+
+        The manifest is created in the specified directory with the filename 'tag_manifest.csv'.
+        Each row in the CSV contains two columns: bucket name and object key.
+
+        Args:
+            write_directory (str): Directory path where the manifest file will be created
+            delete_list (List[str]): List of object keys to be included in the manifest
+
+        Returns:
+            str: Full path to the created manifest file
+
+        Raises:
+            SweeperProcessorError: If the manifest file cannot be created due to
+                file system permissions, disk space, or other I/O errors
+        """
+        csv_formatted_list: List[List[str]] = []
+        for obj_key in delete_list:
+            csv_formatted_list.append([self.bucket_name, obj_key])
+
+        file_path: str = os.path.join(write_directory, "tag_manifest.csv")
```

Comment:
Could we take in a full file path to the manifest CSV? This way callers don't have to create multiple directories if they want to creates multiple tag manifest files